apps/designer/app/canvas/hovered-instance-connector.ts

```diff
@@ -0,0 +1,108 @@
+import { useEffect } from "react";
+import debounce from "lodash.debounce";
+import { type Instance, idAttribute } from "@webstudio-is/react-sdk";
+import { utils } from "@webstudio-is/project";
+import { publish, subscribe } from "~/shared/pubsub";
+import { useRootInstance } from "~/shared/nano-states";
+import { subscribeScrollState } from "~/shared/dom-hooks";
+
+const startHoveredInstanceConnection = (rootInstance: Instance) => {
+  let hoveredElement: undefined | Element = undefined;
+
+  // debounce is used to avoid laggy hover because of iframe message delay
+  const publishHover = debounce((element: Element) => {
+    const id = element.getAttribute(idAttribute) ?? undefined;
+    if (id === undefined) {
+      return;
+    }
+    const instance = utils.tree.findInstanceById(rootInstance, id);
+    if (instance === undefined) {
+      return;
+    }
+    // @todo combine hover events
```

Comment:
why actually? rect can change without instance change, so they are independent ...

Reply:
Logic says they were always send together. Then we can use only one to update hovered state and slightly simplify interaction with designer.

Reply:
I think the logic will change later with the multiplayer if instance rect can change under your mouse lol